src/org.xtuml.bp.integrity/src/org/xtuml/bp/integrity/generator/Generator.java

```diff
@@ -0,0 +1,241 @@
+        }
+        checkReferentialIntegrity();
+    }
+ 
+    /*
+     * The flow of this function is: 
+     * - Call xtumlmc_build xtuml_integrity -i <model folder> -o <model file> > integrity.txt
+     * - Display integrity.txt
+     */
+    private static void checkReferentialIntegrity() {
+
+        IPath path;
+        destPath = "";
+        List<String> modelsDir = new ArrayList<String>();
+        final IProject[] projects = ResourcesPlugin.getWorkspace().getRoot().getProjects();
+        // Build an array of string paths to model folders.
+        for ( IProject project: projects ) {
+            String projPath = project.getLocation().toOSString();
+            IPath modelspath = new Path(projPath + File.separator + "models");
+            if (modelspath.toFile().exists()) {
+                modelsDir.add("-i");
+                modelsDir.add(modelspath.toOSString());
+                if ( "" == destPath ) {
+                    // only set these up on the first project found
+                    path = new Path(projPath + File.separator + DOC_DIR + File.separator);
+                    if (!path.toFile().exists()) {
+                        path.toFile().mkdir();
+                    }
+                    destPath = path.toOSString();
+                    firstProject = project;
+                }
+            }
+        }
+        if ( "" != destPath ) {
+            ProgressMonitorDialog pmd = new ProgressMonitorDialog(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell());
+            pmd.setCancelable(true);
+            pmd.create();
+            IProgressMonitor monitor = pmd.getProgressMonitor();
+            
+            try {
+                // Proceed with actually running integrity on the model
+                IWorkbenchPage page = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage();
+                String id = IConsoleConstants.ID_CONSOLE_VIEW;
+                IConsoleView view = (IConsoleView) page.showView(id);
+                view.display(myConsole);
+
+                pmd.run(true, true, new IRunnableWithProgress() {
+
+                    public void run(IProgressMonitor monitor)
+                            throws InvocationTargetException,
+                            InterruptedException {
+
+                        int steps = 3;
+                        int curStep = 1;
+
+                        monitor.beginTask("Check Referential Integrity", steps);
+
+                        while (curStep <= steps) {
+                            if (monitor.isCanceled()) {
+                                InterruptedException ie = new InterruptedException("User cancelled the operation");
+                                throw ie;
+                            }
+
+                            try {
+                                switch (curStep) {
+                                case 1:
+                                    monitor.subTask("Accumulating model data for workspace");
```

Comment:
It makes a big difference.  :)  It means that the progress bar shows a fraction of progress while they wait for the real work rather than nothing and then all.